### PR TITLE
Added Cohort Route for weeks and activities

### PR DESCRIPTION
**Pull request by Forrestdarabian**
# Description

Created a cohort.routes.ts file for cohort entities id's, week's and activities

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
cohort route for weeks and activities

**Comment by wSedlacek** [comment]
[[Child] - View Current Activity](https://trello.com/c/yuJUJXrS/32-child-view-current-activity)

**Review by wSedlacek — APPROVED** [review]
Looks good to me

File: src/middleware/validate/validate.middleware.ts [hunk 1]
```diff
@@ -30,27 +34,37 @@ const Validation: Middleware = () => async (req, res, next) => {
...
+        if (req.path === '/payment/subscribe')
             req.subscribe = (await transformAndValidate(SubscribeDTO, req.body)) as SubscribeDTO;
 
         //Validates and transforms canon request objects prior to routing
-        if (req.path === '/canon' && req.method === 'POST')
+        if (req.path === '/canon')
             req.addCanon = (await transformAndValidate(AddCanonDTO, req.body)) as AddCanonDTO;
 
+        //Validates and transforms cohort request objects prior to routing
+        if (req.path.match(/\/cohort\/list(\/\d*)?/))
```
Comment (EricBanker12) [inline src/middleware/validate/validate.middleware.ts:49]: regex should be commented with an example of the pattern it should match

File: README.md [hunk 2]
```diff
@@ -63,15 +63,15 @@ To get the server running locally:
...
+| GET    | `/children/list`        | adult users    | Returns a list of child accounts associated with logged in parent. |
+| GET    | `/children/list/:id`    | adult users    | Returns specified child account.                                   |
+| PUT    | `/children/list/:id`    | adult users    | Updates specified child account.                                   |
+| DELETE | `/children/list/:id`    | adult users    | Deletes specified child account.                                   |
 | GET    | `/children/me`          | child users    | Returns logged in child.                                           |
 | GET    | `/children/preferences` | child users    | Returns child's preferences.                                       |
 | GET    | `/children/parent`      | child users    | Returns the child's parent.                                        |
 | POST   | `/children/`            | adult users    | Adds a new child account.                                          |
 | POST   | `/children/:id/login`   | adult users    | Switches from parent to designated child account.                  |
```
Comment (EricBanker12) [inline README.md]: new routes need to be added

File: README.md [hunk 3]
```diff
@@ -103,6 +113,20 @@ To get the server running locally:
...
+
+---
+
+```
+{
+  id: NUMBER
+  name: STRING
+  week: NUMBER
+  activity: STRING
+
```
Comment (EricBanker12) [inline README.md]: children array _may_ need to be added

**Pull request merged**